### ISSUE TITLE
Hang when attempting to message a PR670 under Linux

### ISSUE THREAD

**Issue by tomwallis** [psychopy/psychopy#1698]
I'm running the current psychopy dev version (v 1.90.0) under Ubuntu Linux 16.04.

I'm trying to run a calibration using a PR670 spectroradiometer. Running anything in monitor center causes the app to hang (greys out; requires force quit). This also happens under the current Standalone App (v 1.85.3). 

I've managed to trace the hang to line 391 of /hardware/pr.py ([here](https://github.com/psychopy/psychopy/blob/76e210cfe2487e356e7d0fe81297824a1073ceb8/psychopy/hardware/pr.py#L391)). Basically, on my system, I connect to the device via serial port just fine, but the first time a message is sent ([self.sendMessage](https://github.com/psychopy/psychopy/blob/76e210cfe2487e356e7d0fe81297824a1073ceb8/psychopy/hardware/pr.py#L373)) the whole thing hangs. In this case it's specifically on the "P" of "PHOTO" in "Sending command 'PHOTO' to /dev/ttyACM0". Unfortunately this hangs PyCharm's debugger, so I haven't been able to get more specific information about what's going wrong.

* I've checked that `/dev/ttyACM0` does indeed correspond to the PR670 (via `/dev/serial/by-path/`). 
* I haven't installed any special drivers on this system. Perhaps I'm missing a driver for the PR670, but I was unable to find a reference for one under Linux on the CD that Photo Research distribute with the device. 
* I can "initialize" the PR670 under PsychToolbox but then it times out for any further commands (suggesting it's a system problem).
* Can anyone verify that the calls for the PR655 are the same as for the PR670? PsychoPy lumps them together in the same object class, so I assume someone tested that both work. However, PTB has separate calls for them.

Does anyone have any suggestions?

**Comment by peircej** [comment]
Yes, I can verify that the calls work fine for both devices (but I've never used them from ubuntu). My memory of USB devices on ubuntu was that you have this nasty business of having to arrange permissions so you can write to the device. Did you go through that?

**Comment by peircej** [comment]
If you are successfully detecting the device as a serial device then drivers shouldn't be necessary

**Comment by tomwallis** [comment]
Ok, it looks like this was indeed a permissions problem on Ubuntu. Running the [`psychLinuxConfiguration`](https://github.com/kleinerm/Psychtoolbox-3/blob/master/Psychtoolbox/PsychLinuxConfiguration.m) file of the Psychtoolbox (which among other things, adds the user to a `dialout` group that allows serial port access) has fixed the problem in PsychoPy.

It may be worth checking out what Mario is doing in that script and porting something similar to PsychoPy, or at least probing if the user has sufficient permissions. 

If I were to look into porting this over:

1. How does that work with licensing (if I'm basically copying Mario's code)?
1. Where should such a routine be put within PsychoPy? Ideally it would run when PsychoPy standalone opens the first time, to check that the user has sufficient permissions.

**Comment by tomwallis** [comment]
A simple, hacky solution would be to make people set up their hardware with PTB under Octave first. Then we completely rely on Mario, and also get changes he adopts. I'm not sure if that's super cheeky, though.

e.g. after installing psychopy on linux, a message says: 

Please install ptb3-octave from the neurodebian distro, then

```
octave --no-gui
psychLinuxConfiguration
```

**Comment by peircej** [comment]
1. copying the code is no problem - we just have to keep the license for the code clearly pasted.
2. I'm sure the actual script functions aren't hard to port over. The issue here is that the user probably needs to run the config script as sudo. We don't usually ask users to run the app as sudo. We could certainly have our own config script and tell people to run that with sudo as needed. I imagine the apt-get install procedure could run the script too  (assuming people used apt-get instead of pip install) and that would be the cleanest solution. But maybe both.

I'm not a fan of requiring users to install octave and PTB just for the sake of setting up USB permissions.
